### ISSUE TITLE
Impossible to launch directly XEX files

### ISSUE THREAD

**Issue by Darknior** [libretro/libretro-atari800#63]
Hi

With this commit you have add XEX files to the emulator and it's perfect because i use them. 

https://github.com/libretro/libretro-atari800/commit/6b4c2b2991fec491da7aee2be32ea765d4fe8f3e

But it's impossible to launch them directly, Every time i try to launch a XEX file i fall on the SELF TEST screen.

![maxresdefault](https://user-images.githubusercontent.com/3703317/101393680-bfa76c80-38c7-11eb-8378-8a4b843e8267.jpg)

I must open the old bleu emulator menu, and use RUN ATARI PROGRAM manually.  And them the game working fine.

![Atari800-Configuration-Screen-Screenshot](https://user-images.githubusercontent.com/3703317/101393878-0d23d980-38c8-11eb-83d2-461e8674aa09.png)

I don't have this problem with ATR files for exemple. They start fine to game.
Is it possible to easily fix it please ?

Thanks a lot for all your work on this great emulator !!!

**Comment by raphkoster** [comment]
.XEX files are single executables. They cannot boot the machine unless there is a DOS disk present, usually. The way you would run these on a real Atari would be by booting a DOS disk, going to the DOS menu, and choosing "Run a program" or similar (depends on the specific DOS version what the language is).

.ATR files are disk images. They have a Disk Operating System on them that boots the machine.

(See https://retropie.org.uk/docs/Atari-800-and-5200/#typical-file-formats)

The fact that you are getting the Self-Test screen means that the emulator thinks you are running an XL/XE machine, and have turned off BASIC, and do not have a disk in the drive.

What happens when you try running a .BIN that you know is not a cartridge? Or a .COM file?

**Comment by Darknior** [comment]
Yes i know this doc, but some other emu launch XEX files. Maybe they launch a DOS disk alone when detect an XEX file.
But i understand why it don't works fine. I can use ATR and CAR files without any problem. 
I think you are right i would not use XEX files.
Thanks a lot

**Comment by raphkoster** [comment]
I just haven't tested other standalone executables personally, it doesn't mean it can't be done. If the emulator loads standalone executables with other file extensions, it should work with .xex files. An .xex file is exactly the same as a .exe or a .com, on the Atari platform -- a binary executable.

Have you tried loading other binary executables?

**Comment by Darknior** [comment]
Yes i've try to load some other from the blue menu and they works all fine from it.
But like you explain me it's impossible to launch them from Windows directly.

And finally i have some homebrew only on XEX format :(

Montezuma 2020
http://a8.fandal.cz/detail.php?files_id=7864
https://atariage.com/forums/topic/308584-montezumas-redux/

Bomb Jack
http://www.atarimania.com/game-atari-400-800-xl-xe-bomb-jack_21390.html

Popeye
https://atariage.com/forums/topic/295007-popeye-hack-whaaat-again/page/4/#comments

And i think some other are existing ...

**Comment by raphkoster** [comment]
Can you check your atari800.cfg and see if you have something in the cartridge? Based on what i am reading, if you do not have a cartridge in, it should load the .xex directly.

**Comment by Darknior** [comment]
You are right. I've made so many test this evening and record some bad options. I've removed the VFG file and now all the games are working fine. Really sorry for that.
The only one that show me an error is BOMB JACK with the 320 Kb memory problem. But it works if i change it in settings and reload it.

The only game that give me a black screen for now and i don't found why ? Is this incredible Demo

https://atariage.com/forums/topic/298914-unicorns-season-prince-of-persia-for-the-a8

You have really made a great work on this emulator because now it's really easy to use :)
When i've try it 3 years before ... it was really horrible to make one game working fine.
Thanks
